config/vite/vite.config.mts

```diff
@@ -0,0 +1,29 @@
+/// <reference types="vitest" />
+
+import { defineConfig } from 'vite'; // eslint-disable-line import/no-extraneous-dependencies -- this is a dev-only dependency
+import { fileURLToPath } from 'node:url';
+
+// eslint-disable-next-line import/no-default-export -- default export is a Vite requirement
+export default defineConfig({
+  build: {
+    emptyOutDir: true,
+    lib: {
+      entry: fileURLToPath(new URL('/src/RollbarClient.mts', import.meta.url)),
+      fileName: 'RollbarClient',
+      formats: ['es'],
+    },
+    minify: 'esbuild',
+    modulePreload: false,
+    outDir: 'dist',
+    sourcemap: true,
+    target: 'modules',
+  },
+  test: {
+    clearMocks: true,
+    coverage: {
+      provider: 'c8',
+      reportsDirectory: './vitestCodeCoverage',
+    },
+    environment: 'happy-dom',
+  },
```

Comment:
This is the Vitest-specific section (`test`) of the configuration. Look how easy this is to setup for building and unit tests in one file! And this is in "library mode" which is the less focussed-on case. App builds should be even more straight forward to setup than this.